### PR TITLE
Improved test coverage for terms of use (#2475)

### PR DESCRIPTION
**Pull request by Wiezzel**

### CODE REVIEW

File: tests/golem/test_terms.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+from golem.model import GenericKeyValue
+from golem.terms import TermsOfUse
+from golem.tools.testwithdatabase import TestWithDatabase
+
+
+class TestTermsOfUse(TestWithDatabase):
+
+    def test_are_terms_accepted_no_entry(self):
+        self.assertFalse(TermsOfUse.are_terms_accepted())
```
Comment (Krigpl) [inline tests/golem/test_terms.py:12]: This and all other `self.assert*`s can be replaced with simple `assert not bool` etc which are clearer

File: tests/golem/test_terms.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        self.assertTrue(TermsOfUse.are_terms_accepted())
+
+    def test_accept_terms_right_version(self):
+        GenericKeyValue.create(
+            key=TermsOfUse.TERMS_ACCEPTED_KEY,
+            value=TermsOfUse.TERMS_VERSION)
+        TermsOfUse.accept_terms()
+        self.assertTrue(TermsOfUse.are_terms_accepted())
+
+    @patch('golem.terms.get_golem_path', return_value='/tmp')
```
Comment (Krigpl) [inline tests/golem/test_terms.py]: use `self.tempdir`, `/tmp` won't work on Windows for instance

**Review by Krigpl — CHANGES_REQUESTED** [review]
lgtm, just the temporary dir thing

File: tests/golem/test_terms.py (hunk 1 above)
Comment (Wiezzel) [reply tests/golem/test_terms.py:12]: I am not a fan of using plain assertions in unit tests. `unittest.TestCase.failureException` could be changed to a different exception class than `AssertionError` and then we get different test outcomes. I advocate using plain `assert` only in core code.

File: tests/golem/test_terms.py (hunk 2 above)
Comment (Wiezzel) [reply tests/golem/test_terms.py]: Fixed.

File: tests/golem/test_terms.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    @patch('golem.terms.get_golem_path')
+    def test_show_terms(self, get_golem_path):
+        get_golem_path.return_value = self.tempdir
+        content = """
+        GOLEM TERMS OF USE
+        ==================
+        Bla bla bla bla
+        """
+        os.makedirs(self.new_path / 'golem', exist_ok=True)
+        with open(self.new_path / 'golem' / 'TERMS.html', mode='w') as terms:
```
Comment (Krigpl) [inline tests/golem/test_terms.py]: Actually that path should also be a constant in the `Terms` class

Comment (Wiezzel) [reply tests/golem/test_terms.py]: Good idea.

**Review by Krigpl — APPROVED** [review]

**Pull request merged**